crates/ide/src/file_structure.rs

```diff
@@ -160,7 +160,10 @@ fn structure_node(node: &SyntaxNode) -> Option<StructureNode> {
                 let label = match target_trait {
                     None => format!("impl {}", target_type.syntax().text()),
                     Some(t) => {
-                        format!("impl {} for {}", t.syntax().text(), target_type.syntax().text(),)
+                        format!("impl {}{} for {}", 
+                        it.excl_token().map(|x| x.to_string()).unwrap_or_default(), 
+                        t.syntax().text(), 
+                        target_type.syntax().text(),)
```

Comment:
The change looks correct, but please format the code to be a bit more readable:

```suggestion
                        format!("impl {}{} for {}", 
                            it.excl_token().map(|x| x.to_string()).unwrap_or_default(), 
                            t.syntax().text(), 
                            target_type.syntax().text(),
                        )
```